src/flinkStatementResultsManager.test.ts

```diff
@@ -261,81 +187,219 @@ describe("FlinkStatementResultsManager", () => {
-    assert.equal(flinkSqlStatementsApi.updateSqlv1Statement.callCount, 5);
+    await vm.stopStatement();
+    assert.equal(ctx.flinkSqlStatementsApi.updateSqlv1Statement.callCount, 5);
   });
 
   it("should stop polling when statement is not results viewable", async () => {
-    await createResultsManagerWithResults();
-
-    assert.ok(manager["_pollingInterval"] as NodeJS.Timeout);
+    assert.ok(ctx.manager["_pollingInterval"] as NodeJS.Timeout);
 
     const nonViewableStatement = new FlinkStatement({
-      ...mockStatement,
+      ...ctx.statement,
       status: {
-        ...mockStatement.status,
+        ...ctx.statement.status,
         phase: Phase.FAILED,
         detail: "Statement failed",
       },
     });
-    refreshFlinkStatementStub.returns(Promise.resolve(nonViewableStatement));
+    ctx.refreshFlinkStatementStub.returns(Promise.resolve(nonViewableStatement));
 
     // Verify polling was stopped
     await eventually(() => {
-      assert.equal(manager["_pollingInterval"], undefined);
+      assert.equal(ctx.manager["_pollingInterval"], undefined);
     });
   });
 
   it("should handle non-409 errors in StopStatement immediately", async () => {
-    await createResultsManagerWithResults();
-
-    flinkSqlStatementsApi.updateSqlv1Statement.rejects(
+    ctx.flinkSqlStatementsApi.updateSqlv1Statement.rejects(
       createResponseError(500, "Server Error", "test"),
     );
 
-    await manager.handleMessage("StopStatement", {});
+    await vm.stopStatement();
+
+    assert.equal(ctx.flinkSqlStatementsApi.updateSqlv1Statement.callCount, 1);
+  });
 
-    assert.equal(flinkSqlStatementsApi.updateSqlv1Statement.callCount, 1);
+  describe("FlinkStatementResultsViewModel only", () => {
+    describe("schema and columns", () => {
+      it("should create correct column definitions for table view", () => {
+        const columns = vm.columns();
+        assert.deepStrictEqual(Object.keys(columns), ["when_reported", "tempf"]);
+        assert.strictEqual(columns["when_reported"].title(), "when_reported");
+        assert.strictEqual(columns["tempf"].title(), "tempf");
+      });
+
+      it("should get schema correctly", () => {
+        const schema = vm.schema();
+
+        assert.deepStrictEqual(schema, {
+          columns: [
+            {
+              name: "when_reported",
+              type: {
+                nullable: false,
+                precision: 6,
+                type: "TIMESTAMP_WITH_LOCAL_TIME_ZONE",
+              },
+            },
+            {
+              name: "tempf",
+              type: {
+                nullable: false,
+                type: "DOUBLE",
+              },
+            },
+          ],
+        });
+      });
+
+      it("should handle empty schema", () => {
+        vm.schema({ columns: [] });
+        const columns = vm.columns();
+        assert.deepStrictEqual(Object.keys(columns), []);
+      });
+    });
+
+    describe("pagination", () => {
```

Comment:
test(s) combining search + pagination? You know, my favorite topic!

Reply:
I can do better job of organizing these `describe`s and `it`s but anyway, test over search + pagination is here --`should filter and then paginate results based on search query`.